### PR TITLE
feat: add image pull secrets to service account

### PR DESCRIPTION
**Pull request by Jaxwood**
**Reason for Change**:
Introduces imagePullSecrets to the admin service account.
This allows to pulling images from private registries
where authentication is required.

**Is this a deployment yaml update?**
no

**Are you making changes to the Helm chart?**
yes

**Requirements**

- [x] squashed commits
- [x] included documentation
- [x] added unit tests and e2e tests (if applicable).

**Issue Fixed**:
no

Does this change contain code from or inspired by another project? If so, did you notify the maintainers and provide attribution?

- [ ] yes
- [x] no

**Notes for Reviewers**:

Didn't add any tests. If you have tests for your helm charts I'm of course willing to add.

### CODE REVIEW

File: manifest_staging/charts/workload-identity-webhook/templates/azure-wi-webhook-admin-serviceaccount.yaml [hunk 1]
```diff
@@ -1,3 +1,7 @@
+{{- if .Values.imagePullSecrets }}
+imagePullSecrets:
+{{ toYaml .Values.imagePullSecrets | nindent 2 }}
```
Comment (aramase) [inline manifest_staging/charts/workload-identity-webhook/templates/azure-wi-webhook-admin-serviceaccount.yaml:3]:
```suggestion
{{- toYaml .Values.imagePullSecrets | nindent 2 }}
```

File: third_party/open-policy-agent/gatekeeper/helmify/kustomize-for-helm.yaml [hunk 2]
```diff
@@ -88,3 +88,11 @@ webhooks:
   failurePolicy: HELMSUBST_MUTATING_WEBHOOK_FAILURE_POLICY
   name: mutation.azure-workload-identity.io
   objectSelector: HELMSUBST_MUTATING_WEBHOOK_OBJECT_SELECTOR
+---
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: azure-wi-webhook-admin
+  namespace: azure-workload-identity-system
+HELMSUBST_SERVICEACCOUNT_IMAGEPULLSECRETS: ""
```
Comment (aramase) [inline third_party/open-policy-agent/gatekeeper/helmify/kustomize-for-helm.yaml:97]:
```suggestion
HELMSUBST_SERVICEACCOUNT_IMAGE_PULL_SECRETS: ""
```

File: third_party/open-policy-agent/gatekeeper/helmify/kustomize-for-helm.yaml [hunk 3]
```diff
@@ -88,3 +88,11 @@ webhooks:
...
+---
```
Comment (aramase) [inline third_party/open-policy-agent/gatekeeper/helmify/kustomize-for-helm.yaml:98]:
```suggestion

```

File: third_party/open-policy-agent/gatekeeper/helmify/replacements.go [hunk 4]
```diff
@@ -28,4 +28,10 @@ var replacements = map[string]string{
 	`HELMSUBST_MUTATING_WEBHOOK_OBJECT_SELECTOR`: `{{- toYaml .Values.mutatingWebhookObjectSelector | nindent 4 }}`,
 
 	`HELMSUBST_MUTATING_WEBHOOK_ANNOTATIONS: ""`: `{{- toYaml .Values.mutatingWebhookAnnotations | nindent 4 }}`,
+
+	`HELMSUBST_SERVICEACCOUNT_IMAGEPULLSECRETS: ""`:
+`{{- if .Values.imagePullSecrets }}
+imagePullSecrets:
+{{ toYaml .Values.imagePullSecrets | nindent 2 }}
```
Comment (aramase) [inline third_party/open-policy-agent/gatekeeper/helmify/replacements.go:35]:
```suggestion
{{- toYaml .Values.imagePullSecrets | nindent 2 }}
```

File: third_party/open-policy-agent/gatekeeper/helmify/kustomize-for-helm.yaml [hunk 5]
```diff
@@ -88,3 +88,10 @@ webhooks:
   failurePolicy: HELMSUBST_MUTATING_WEBHOOK_FAILURE_POLICY
   name: mutation.azure-workload-identity.io
   objectSelector: HELMSUBST_MUTATING_WEBHOOK_OBJECT_SELECTOR
+---
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: azure-wi-webhook-admin
+  namespace: azure-workload-identity-system
+HELMSUBST_SERVICEACCOUNT_IMAGE_PULL_SECRETS: ""
```
Comment (aramase) [inline third_party/open-policy-agent/gatekeeper/helmify/kustomize-for-helm.yaml:97]: nit: add new line

Comment (Jaxwood) [reply third_party/open-policy-agent/gatekeeper/helmify/kustomize-for-helm.yaml]: @aramase thanks - added

**Comment by Jaxwood** [comment]
> /azp run Azure Workload Identity CI

@aramase I'm sorry - but I forgot to sign my previous commit and had to force push

**Comment by aramase** [comment]
> @aramase I'm sorry - but I forgot to sign my previous commit and had to force push

No worries! i'll retrigger the CI

**Review by aramase — APPROVED** [review]
Thank you for the PR, @Jaxwood! 🎉 

lgtm

**Pull request merged**